### ISSUE TITLE
getIsAuthorized returning true even when authentication is false

### ISSUE THREAD

**Issue by Aeseir** [damienbod/angular-auth-oidc-client#155]
Heya,

Great package here you got, love it to bits.

I have recently implemented a guard  that checks if user is authenticated before passing through the user.
Overall works great, except when user's token has expired.

```
canActivateChild(
    route: ActivatedRouteSnapshot,
    state: RouterStateSnapshot
  ): Observable<boolean> | Promise<boolean> | boolean {

    return this.oidcSecurityService.getIsAuthorized()
      .map((isAuthorized: boolean) => {
        console.info(isAuthorized);
        if (isAuthorized === true) {
          return true;
        }

        this.oidcSecurityService.authorize();
        return false;
      });
  }
```

What i noticed is that getIsAuthorized() returns a true, yet when it submits the token, the api call results in a 401 (Unauthorized).

Reading the documentation getIsAuthorized() returns true when client and user are authenticated.

1. Is it possible to have the client to be authenticated and user to be unauthenticated and have the method still throw true?

2. If yes to 1 -> What is the proper method or combination that can be used to check if client and user are authenticated?

**Comment by damienbod** [comment]
@Aeseir 
I'll try to reproduce this using a guard. Should not be possible, but I need to check. There might be a timing issue when initializing the app. The providers and the guards don't wait for the APP_INIT

**Comment by damienbod** [comment]
@Aeseir I think I found the bug. Want to do some tests first. I the init:

 const isAuthorized = this.oidcSecurityCommon.isAuthorized;
        if (isAuthorized) {
            this.setIsAuthorized(isAuthorized);

            // Start the silent renew
            this.runTokenValidation();
        }
so I use the old value, even if the token is old. So for a race condition, the is authorized is true.

**Comment by damienbod** [comment]
https://github.com/damienbod/angular-auth-oidc-client/issues/156

**Comment by Aeseir** [comment]
Awesome, I thought i was going crazy after the 10 hour coding session. Let me know if you need any further information to help you with this.

**Comment by damienbod** [comment]
fixed in release 4.0.0

**Comment by Aeseir** [comment]
isntalled it, ran it, tested it, works. Legendary, owe you a beer
